### ISSUE TITLE
bug: security scheme name starting with `X-` not being added to SecuritySchemes map

### ISSUE THREAD

**Issue by TristanSpeakEasy** [pb33f/libopenapi#54]
If a OpenAPI document defines a security scheme name starting with `X-` it is not showing up in `model.Components.SecuritySchemes`

for example:

```yaml
securitySchemes:
    OAuth:
      flows:
        authorizationCode:
          authorizationUrl: "https://example.com/v1/oauth/authorize"
          refreshUrl: "https://example.com/v1/oauth/token"
          scopes:
            account.manage: This scope grants permissions to perform read/edit/delete actions on Account data
            account.view: This scope grants permissions to perform read only actions on Account data
            openid: This scope grants permissions that enable SSO by granting an id token JWT that stores account data. Not used in v1/account endpoints
          tokenUrl: "https://example.com/v1/oauth/token"
      type: oauth2
    X-API-Key:
      in: header
      name: X-API-Key
      type: apiKey
```

`X-API-Key` can't be found

**Comment by daveshanley** [comment]
Another extension glitch I think, Will address ASAP.

**Comment by daveshanley** [comment]
Resolved in `v0.4.8`

**Comment by rpmccarter** [comment]
Hello! Seems like this is fixed for `X-`, but is still broken for `x-`. I've opened #188, which adds test cases for both cases, and I'm hoping someone can help me fix the underlying bug. Thanks!
